### ISSUE TITLE
RaceDay Scoring showing old result after edit

### ISSUE THREAD

**Issue by louking** [louking/tm-csv-connector#48]
This was during testing for #44

I'm testing RDS time machine interface using a file. I simulated the case where the entry at the time machine was incorrect for one runner. For this simulation the bib number 1691 should have been 1694. The file was edited with 1694, and in RDS the raw read for 1691 was deleted, then the time machine stream was replayed. The incorrect 1691 result persists in the overall report and at RunSignUp's results for this race. Is there another operation I missed to delete the 1691 result?

**Comment by louking** [comment]
Clearing raw read did take care of removing the errant result in RDS as when I cleared the raw reads the leaderboard did indeed get cleared and 1691 is not in RDS overall results report now. Unfortunately I still see 1691 in the runsignup results. I've refreshed https://runsignup.com/Race/Results/116536#resultSetId-431179;perpage:100 for several minutes and his result persists. I know I can manually delete the result on RSU, but I'm not sure why it doesn't get cleaned up when syncing with RDS.

![image](https://github.com/louking/tm-csv-connector/assets/875138/3897bc66-8ab6-450f-b051-869e1983ef03)


![image](https://github.com/louking/tm-csv-connector/assets/875138/0bbc22e1-9d39-4015-9480-51b2e49d43f0)


Also I want to know if there's a less heavy-handed way to clear the results associated with a particular raw read.

I'm considering the case where we will be using Time Machine as backup for a chip system. The scenario I'm worried about is a mistyped bib number on the Time Machine causing a result for a runner who hasn't finished yet. When we discover this we want to be able to fix the results. In this case, we would prefer not to edit the time machine file. Would ignoring the errant raw read in RDS take care of it, and sync the results properly with RSU?

**Comment by louking** [comment]
per Kevin at RSU:

_It looks like when you delete a raw read, you will need to select the recalc button in order for results to be updated in RDS. I did just do a quick test and it does look like when read is deleted and you recalc (so results are updated in RDS) the **autosave for some reason is not getting updated as well**. I'll need to create an issue for our development team to look into this. I do also think the recalc should be happening automatically as well so I'll create an issue for them on that as well. _ [emphasis added]

This does not help get RSU results squared away as mentioned. Subsquently went to Scored Events and did **Save**, which corrected the RSU results.

**Comment by louking** [comment]
Workaround for fully updating a confirmed entry:

* for any bib number which was removed, delete associated raw reads at Raw Reads view
* under *RAW READS* on dashboard, click **RECALC**, then **RECALCULATE**
* under Scored Events view, click **Save**

**Comment by louking** [comment]
RSU claims this is fixed in RDS 4.0.23: https://info.runsignup.com/2024/01/04/new-read-filtering-options-announcer-features-and-more-in-raceday-scoring-v4-0-23/

**Comment by louking** [comment]
It doesn't look like RDS 4.0.23 fully solved this issue.

I did the following test:

- added bib 1700 to the file
- updated file by editing 1700 to 1701
- deleted raw read 1700
- replay stream

The report in RaceDay Scoring is correct

![image](https://github.com/louking/tm-csv-connector/assets/875138/27a9b29b-ed21-4f8f-9949-a8db90bdb4c8)

however the results on RunSign show the errant 1700

![image](https://github.com/louking/tm-csv-connector/assets/875138/9a4a74e9-b271-43d8-96e1-a5cd20b92544)

I still need to go to the Scored Events view and click Save in order for the results to be saved to RSU correctly. Below is after that operation.

![image](https://github.com/louking/tm-csv-connector/assets/875138/aa9e203a-f343-419f-a445-2db51b51efd4)

**Comment by louking** [comment]
From Matt Avery (2/15/24):

_We have corrected the issue where participants would remain in progress after clearing their reads in the latest update. [v4.0.33]_

**Comment by louking** [comment]
I don't see that this is fixed.

I had some issues on setup. These times are approximate. This isn't my complaint, but is the background

```
2024-03-11 17:01 edit 1701 to be 1700 in csv file
2024-03-11 17:12 RDS start read connector
2024-03-11 17:13 delete 1701 raw read
2024-03-11 17:13 RDS start autosaves
2024-03-11 17:13 noticed no results on RSU
2024-03-11 17:17 delete all raw results, replay stream
2024-03-11 17:17 still no results on RSU
2024-03-11 17:19 edit overall report, save
2024-03-11 17:19 results now on RSU
```

I was more careful on the times below.

```
2024-03-11 17:24 edit 1700 to be 1701 in csv file
2024-03-11 17:25 delete 1700 raw read
2024-03-11 17:26 1700 still on RSU results
2024-03-11 17:26 RDS replay stream
2024-03-11 17:27 1700 still on RSU results. screenshot https://i.imgur.com/s49mQbm.png
2024-03-11 17:30 RDS scored events > save
2024-03-11 17:30 1701 is now on RSU results. screenshot https://i.imgur.com/D5Bm1XR.png
2024-03-11 17:32 RDS get help log sent
```

**Comment by louking** [comment]
This was resolved with RDS v4.0.34

I reproduced the problem with v4.0.33, and verified the fix with v4.0.34. Thanks!

```
2024-03-14 05:14 RDS v4.0.33
2024-03-14 05:14 edit 1700 to 1701 in csv file
2024-03-14 05:15 RDS replay stream
2024-03-14 05:15 RDS delete raw read 1700
2024-03-14 05:16 (wait > 30 sec)
2024-03-14 05:17 RSU 1700 remains in result set, but 1701 is seen [see previous screenshots]
2024-03-14 05:17 RDS scored events save
2024-03-14 05:18 RSU 1700 not seen in result set

2024-03-14 05:18 install RDS [v4.0.34]
2024-03-14 05:20 RDS start read connector
2024-03-14 05:20 RDS start autosaves
2024-03-14 05:21 edit 1701 to 1700 in csv file
2024-03-14 05:21 RDS replay stream
2024-03-14 05:22 RDS delete raw read 1701
2024-03-14 05:22 (wait > 30 sec)
2024-03-14 05:23 RSU shows 1700 and not 1701 (i.e., fixed). https://i.imgur.com/Yuqt657.png
```
